lib/actions/upload.js

```diff
@@ -11,22 +11,36 @@ const Sftp = require('../Sftp');
  * @param cfg configuration that is account information and configuration field values
  */
 exports.process = async function processAction(msg, cfg) {
+  this.logger.info('Connecting to sftp server...');
   const sftp = new Sftp(this.logger, cfg);
   await sftp.connect();
 
   const result = {
     results: [],
   };
   const dir = cfg.directory || '/';
+  // eslint-disable-next-line no-nested-ternary
+  const filename = msg.body.filename
+    ? (Object.keys(msg.attachments).length > 1 ? msg.body.filename.split('.')[0] : msg.body.filename)
+    : null;
```

Comment:
extract this code to method with understandable name, right now it looks over complicated

Reply:
Other way is complicated if-else structure - ternary operator is simpier to read